sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/SessionStorage.java

```diff
@@ -0,0 +1,207 @@
+package com.microsoft.appcenter.persistence;
+
+import android.support.annotation.VisibleForTesting;
+import android.support.annotation.WorkerThread;
+
+import com.microsoft.appcenter.utils.AppCenterLog;
+import com.microsoft.appcenter.utils.storage.StorageHelper;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.UUID;
+
+import static com.microsoft.appcenter.AppCenter.LOG_TAG;
+
+/**
+ * Persistent session history.
+ */
+public class SessionStorage {
+
+    /**
+     * Key used in storage to persist sessions.
+     */
+    private static final String STORAGE_KEY = "sessions";
+    /**
+     * Maximum number of sessions to persist the state.
+     */
+    private static final int STORAGE_MAX_SESSIONS = 10;
+    /**
+     * Separator used for persistent storage format.
+     */
+    private static final String STORAGE_KEY_VALUE_SEPARATOR = "/";
+    /**
+     * Singleton.
+     */
+    private static SessionStorage sInstance;
+    /**
+     * Past and current session identifiers sorted by session starting timestamp (ascending).
+     */
+    private final NavigableMap<Long, SessionInfo> mSessions = new TreeMap<>();
+
+    /**
+     * App launch timestamp. TODO we should use the real process start time and not SDK start time.
+     * But there is no Android API to do that it requires executing ps command or reading proc files.
+     * This is used to know minidump files app launch timestamp that are processed after restart.
+     * This is not used for regular managed crashes.
+     */
+    private final long mAppLaunchTimestamp;
+
+    /**
+     * Init.
+     */
+    @WorkerThread
+    private SessionStorage() {
+
+        /* Try loading past sessions from storage. */
```

Comment:
I think we can just break it as Apple did. It is already not quite accurate.

Reply:
Break what?

Reply:
As we talked offline, this is not the one that support backward compatibility. Ignore my comment.